CMakeLists.txt

```diff
@@ -79,7 +79,7 @@ if (CMAKE_COMPILER_IS_GNUCXX)
     # Enable maximum set of warnings. -Wno-sign-compare is required because of
     # problems in gtest. -Wswitch-default would be nice as well, but also causes
     # problems in gtest.
-    add_definitions(-Wall -Wextra -Wformat-security -Wfloat-equal -Wcast-qual -Wconversion -Wlogical-op -Wno-sign-compare)
+    add_definitions(-Wall -Wextra -Wformat-security -Wfloat-equal -Wcast-qual -Wconversion -Wlogical-op -Wno-sign-compare -Wno-error=conversion)
```

Comment:
I would actually just remove this and the -Wconversion in the same line, and maybe add to the comment right above it that -Wconversion would be nice if it didn't cause problems with gtest. I would maybe also add -Wconversion to the PEDANTIC_WARNINGS section just a few lines below.

The combination "-Werror -Wconversion -Wno-error=conversion" is perhaps undesirable since it will still show warning messages for conversion problems but won't break.

For the record, there doesn't appear to be much we can do about this type of thing at present, unless there is an easy way in cmake to manually remove flags just for a specific sub-project/subdirectory.